### ISSUE TITLE
[sync] Error: there is no branch

### ISSUE THREAD

**Issue by ruudk** [git-town/git-town#4132]
### Steps to reproduce

* Create a feature, propose it
* Append branch: feature 2
* Merge the feature 1
* Remove feature 1 branch manually, via git
* Run sync on feature 2
* "Error: there is no branch "ruudk/feature1"
* Fix it by running git town set-parent > set to main
* Run sync again

I think in such a case, where the local parent branch has been removed, git town should detect it and offer a solution. 

Something like: "It looks like you removed the local parent branch feature 1. Please specify the parent branch and run again"

### Command output

```shell
git town config get-parent
ruudk/feature1

git branch -D ruudk/feature1

git town sync
[ruudk/feature2] git fetch --prune --tags

Error: there is no branch "ruudk/feature1"

git town set-parent

Looking for proposal online ... none
Selected parent branch for "ruudk/feature2": main

git town sync
works
```

**Comment by kevgo** [comment]
Good catch, the current behavior definitely isn't the right user experience. I think if a branch gets deleted from a stack, then Git Town should just assume it was shipped or deleted externally and remove it from the stack, i.e. set the parent of the child branches to the parent of the deleted branch. This situation is similar situation to when the tracking branch gets deleted but the local branch still exists.

For reference, here is how `git town sync` behaves in the various scenarios when a branch is listed as a parent branch in metadata but it doesn't exist.

| LOCAL BRANCH  | TRACKING BRANCH | EXPECTED BEHAVIOR |
|---------------|-----------------|-------------------|
| doesn't exist | doesn't exist   | Assume the branch is was shipped or deleted, set the parent branch of its child branches to the parent of the deleted branch. |
| doesn't exist | exists     | Check out the branch locally. This will work because the branch still exists remotely. |
| exists  | doesn't exist | Push the branch to origin. |
| exists  | existed, now marked as deleted | Assume the branch was shipped remotely and delete the local branch if it has no unshipped changes (same behavior as today). |

**Comment by ruudk** [comment]
> Town should just assume it was shipped or deleted externally and remove it from the stack

I think so too

**Comment by kevgo** [comment]
This is released in [v16.7.0](https://github.com/git-town/git-town/releases/tag/v16.7.0).
